### PR TITLE
feat: implementation of single container / client-server through profile-launcher

### PR DESCRIPTION
**Pull request by jim-wang-intel**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-self-checkout/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close

close: #**issue_number**

## Test Instructions if applicable

- git clone this PR
- make build-profile-launcher
- start camera-simulator: `./camera-simulator/camera-simulator.sh`
- for "test" profile:
```console
PIPELINE_PROFILE="test" sudo -E ./run.sh --workload ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0
```
- for "classification" profile:
```console
PIPELINE_PROFILE="classification" RENDER_MODE=1 sudo -E ./run.sh --workload ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0
```

- you can also run `smoke tests` from make file target: 
```console
make run-smoke-tests
```
it will take a while but it should show ALL PASSED for all ovms profiles.

### CODE REVIEW

**Review by brian-intel — APPROVED** [review]

File: configs/opencv-ovms/cmd_client/main.go [hunk 1]
```diff
@@ -98,13 +139,103 @@ func main() {
...
+	// apply all envs from env files if any
+	envList := ovmsClientConf.OvmsServer.readServerEnvs(envFileDir)
+	cmd.Env = append(cmd.Env, envList...)
+	// override envs from the origEnvs
+	cmd.Env = append(cmd.Env, origEnvs...)
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Printf("failed to run the ovms server launch : %v", err)
+		log.Printf("output: %v", string(output))
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply configs/opencv-ovms/cmd_client/main.go:203]: no, for log, we  don't need \n it is added automatically: https://github.com/golang/go/blob/master/src/log/log.go#L239-L241

**Comment by jim-wang-intel** [comment]
rebased to merge the latest main branch and resolve the conflicts

File: run_smoke_test.sh [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+echo "Running gst profile..."
+PIPELINE_PROFILE="gst" sudo -E ./run.sh --workload ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0
+status_code=$?
+verifyStatusCode gst
+# allowing some time to process
+sleep 30
+verifyNonEmptyPipelineLog gst
+teardown
+
+#7. instance_segmentation profile:
```
Comment (vli11) [inline run_smoke_test.sh:113]: you skipped #6....very minor though.

Comment (jim-wang-intel) [reply run_smoke_test.sh:113]: must be Friday, could not count properly :-P

Comment (jim-wang-intel) [reply run_smoke_test.sh]: fixed.

**Review by antoniomtz — COMMENTED** [review]
Not sure what should be the output of the smoke test, but I get this error in an infinite loop:

`No matching models: "bit_64"`

**Comment by jim-wang-intel** [comment]
> Not sure what should be the output of the smoke test, but I get this error in an infinite loop:
> 
> `No matching models: "bit_64"`

that is probably relate to model-downloading issue.  we could debug it next week.

**Comment by vli11** [comment]
the smoke test looks good on my machine after fixing the permission issue on camera simulator.

**Comment by antoniomtz** [comment]
> the smoke test looks good on my machine after fixing the permission issue on camera simulator.

What did you do to make the camera simulator work?, When I start the script, I only see 1 container running and unable to access the rtsp path.

**Comment by jim-wang-intel** [comment]
> > the smoke test looks good on my machine after fixing the permission issue on camera simulator.
> 
> What did you do to make the camera simulator work?, When I start the script, I only see 1 container running and unable to access the rtsp path.

it's about the camera simulator log file owned by root, just sudo delete that one and it should work;  if not please also check the camera-simulator/ directory ownership itself,: it should be owned by your username and group; not by root.

**Comment by brian-intel** [comment]
Smoke tests passing on my NUC

**Review by antoniomtz — APPROVED** [review]
LGTM

**Pull request merged**